crates/router/src/analytics.rs

```diff
@@ -447,6 +452,7 @@ pub mod routes {
         .await
     }
 
+    #[cfg(feature = "v1")]
```

Comment:
Why is this feature flag added only for certain APIs and not for others?
For ex
- It is applied for `get_profile_payment_metrics` and not for `get_merchant_payment_metrics` or `get_org_payment_metrics`

Reply:
I've only added the V1 flag for functions that use fields present only in V1 domain models and are deprecated in V2. Other functions don't need this feature flag since they don't use any such fields.